components/p3a/BUILD.gn

```diff
@@ -1,42 +1,71 @@
-# Copyright (c) 2019 The Brave Authors. All rights reserved.
+# Copyright (c) 2022 The Brave Authors. All rights reserved.
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
 import("//brave/components/p3a/buildflags.gni")
 import("//build/buildflag_header.gni")
+import("//testing/test.gni")
 
 buildflag_header("buildflags") {
   header = "buildflags.h"
-  flags = [ "BRAVE_P3A_ENABLED=$brave_p3a_enabled" ]
+  flags = [
+    "BRAVE_P3A_ENABLED=$brave_p3a_enabled",
+    "P3A_JSON_UPLOAD_URL=\"$p3a_json_upload_url\"",
+    "P3A_CREATIVE_UPLOAD_URL=\"$p3a_creative_upload_url\"",
+    "P2A_JSON_UPLOAD_URL=\"$p2a_json_upload_url\"",
+    "P3A_STAR_UPLOAD_URL=\"$p3a_star_upload_url\"",
+    "P2A_STAR_UPLOAD_URL=\"$p2a_star_upload_url\"",
+    "STAR_RANDOMNESS_HOST=\"$star_randomness_host\"",
+  ]
 }
 
 static_library("p3a") {
   sources = [
     "brave_p2a_protocols.cc",
     "brave_p2a_protocols.h",
-    "brave_p3a_log_store.cc",
-    "brave_p3a_log_store.h",
+    "brave_p3a_config.cc",
```

Comment:
would be great to drop all `brave_` prefixes, if that doesn't take much time before merging, at least in new files

Reply:
yeah, i've been thinking about this too. it's a good move to make

Reply:
will do this tomorrow, will also replace `brave` namespace with `p3a`

Reply:
done